mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlDBRecord.java

```diff
@@ -74,7 +74,12 @@ protected void writeToDB(PreparedStatement stmt, @Nullable Schema.Field field, i
       case Types.DATE:
         Schema schema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
         if (schema.getType().equals(Schema.Type.INT) && !Schema.LogicalType.DATE.equals(schema.getLogicalType())) {
-          stmt.setInt(sqlIndex, record.get(field.getName()));
+          Object fieldValue = record.get(field.getName());
+          if (fieldValue == null) {
```

Comment:
it doesn't let us pass a null value with setInt()? 

It seems error-prone to force each implementation to check for a null value, can this be done in the parent class?

Reply:
Yes, Statement is A SQL type, different drivers can have this differently. This code segment was a miss in the previous change and got detected as part of E2E tests addition in this PR.

Reply:
it looks like there is already logic for this in the parent (https://github.com/data-integrations/database-plugins/blob/f717eb2b1a0a0cc4726b365f11d8bbe19d3d6989/database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java#L293). It's also doing the getNonNullable() logic for the schema.

To avoid duplicate code, it would be better to refactor the writeToDB() method in DBRecord into 2 methods. The first part does the null handling and then calls some new method (i.e writeNonNullToDB()). That way children don't need to duplicate the logic.